File: doc-examples/hibernate-reactive-example-java/src/main/java/example/ManufacturerRepository.java

package example;

import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.reactive.ReactorCrudRepository;
import reactor.core.publisher.Mono;

@Repository
public interface ManufacturerRepository extends ReactorCrudRepository<Manufacturer, Long> {

    Mono<Manufacturer> save(String name);
}

File: doc-examples/hibernate-reactive-example-kotlin/src/main/kotlin/example/ManufacturerRepository.kt
package example

import io.micronaut.data.annotation.Repository
import io.micronaut.data.repository.CrudRepository
import io.micronaut.data.repository.kotlin.CoroutineCrudRepository

@Repository
interface ManufacturerRepository : CoroutineCrudRepository<Manufacturer, Long> {

    suspend fun save(name: String): Manufacturer
}
